CRM/Utils/Date.php

```diff
@@ -2229,4 +2229,57 @@ public static function convertDateToLocalTime($dateObject, $format = 'YmdHis') {
     return $dateObject->format($format);
   }
 
+  /**
+   * Convert a date string between time zones
+   *
+   * @param string $date - date string using $format
+   * @param string $tz_to - new time zone for date
+   * @param string $tz_from - current time zone for date
+   * @param string $format - format string specification as per DateTime::createFromFormat; defaults to 'Y-m-d H:i:s'
+   *
+   * @throws \CRM_Core_Exception
+   *
+   * @return string;
+   */
+  public static function convertTimeZone(string $date, string $tz_to = NULL, string $tz_from = NULL, $format = NULL) {
+    $tz_from = new DateTimeZone($tz_from ?: CRM_Core_Config::singleton()->userSystem->getTimeZoneString());
+    $tz_to = new DateTimeZone($tz_to ?: CRM_Core_Config::singleton()->userSystem->getTimeZoneString());
+
+    if ($tz_from == $tz_to) {
```

Comment:
So I've discovered this line for unknown reasons is problematic on civicrm.org, comparing them as objects. It thinks they are equal even when not. Short answer is maybe using `if ($tz_from->getName() == $tz_to->getName()) {` might be more reliable? Longer version: https://chat.civicrm.org/civicrm/pl/ft8pkwd8gpdiig3foe78ckoo7o

Reply:
It had not occurred to me that this could be different between PHP 7.4 and earlier PHP versions....
the getName() test seems reasonable.

Reply:
Actually probably the most sensible way to to this guard would be to instead do the UF time zone substitution with null and compare the resulting strings before creating any local objects.  Since I'm reasonable certain it's there for Performance Reasons™ and creating the DateTimeZone objects is probably one of the slowest parts of this function....